src/read.c

```diff
@@ -4588,9 +4575,7 @@ static avifResult avifParse(avifDecoder * decoder)
 #if defined(AVIF_ENABLE_EXPERIMENTAL_MINI)
         sawEverythingNeeded = sawEverythingNeeded && (!needsMini || miniSeen);
 #endif
-#if defined(AVIF_ENABLE_GAIN_MAP)
         sawEverythingNeeded = sawEverythingNeeded && (!needsTmap || tmapSeen);
```

Comment:
Nit: Merge this with line 4574 in the new code (`avifBool sawEverythingNeeded = ftypSeen && (!needsMeta || metaSeen) && (!needsMoov || moovSeen);`)